backend/test/deal-observer.test.js

```diff
@@ -74,3 +76,40 @@ describe('deal-observer-backend', () => {
     assert.deepStrictEqual(expected, actual)
   })
 })
+
+describe('deal-observer-backend binary', () => {
+  let pgPool
+  before(async () => {
+    pgPool = await createPgPool()
+    await migrateWithPgClient(pgPool)
+  })
+  beforeEach(async () => {
+    await pgPool.query('DELETE FROM active_deals')
+  })
+  after(async () => {
+    await pgPool.query('DELETE FROM active_deals')
+  })
+  it('then deal observer loop fetches new active deals and stores them in storage', async (t) => {
+    const controller = new AbortController()
+    const { signal } = controller
+    dealObserverLoop(
+      makeRpcRequest,
+      pgPool,
+      undefined,
+      undefined,
+      // The testdata has a total amount of 11 blocks
+      11,
+      0,
+      1,
+      undefined,
+      signal
+    )
+    // Timeout to kill the worker after 1 second
+    let rows
+    setTimeout(() => { if (!signal.aborted) { throw new Error(`Test timed out. Rows inserted ${rows.length}`) } }, 1000)
+    do {
+      rows = (await pgPool.query('SELECT * FROM active_deals')).rows
+    } while (rows.length !== 360)
```

Comment:
Nitpick - feel free to ignore.

Fetching data for all rows is not a very efficient way to count the number of rows in a table. It does not matter in this test, since you are fetching up to 360 rows, but it could become a problem if used elsewhere.

A more efficient solution is to use the SQL aggregate function `COUNT()`.

```js
while (true) {
  const { rows } = await pgPool.query('SELECT COUNT(*) FROM active_deals')
  if (rows[0].count === 360) break
}
```